great_expectations/core/async_executor.py

```
@@ -0,0 +1,133 @@
+        call to submit. This is useful for introducing multithreading while still supporting single threaded execution.
+        This allows configuration to determine whether or not multithreading is used, which is useful when there are
+        concerns that some workflows may not benefit from multithreading or may not be safe to multithread.
+
+        This class is intended to be used as a context manager using the `with` statement.
+
+        Args:
+            concurrency_config: Configuration used to determine whether or not execution is done concurrently with
+                multiple threads. Even if the configuration has concurrency enabled, if max_workers is 1 then all work
+                will be done synchronously (e.g. on the main thread) during the call to submit.
+            max_workers: The maximum number of threads that can be used to execute concurrently. If concurrency is
+                disabled or max_workers is 1, all work will be done synchronously (e.g. on the main thread) during the
+                call to submit. Note that the maximum number of threads is also limited by
+                concurrency_config.max_database_query_concurrency.
+        """
+        # Only enable concurrent execution if it is enabled in the config AND there is more than 1 max worker specified.
+        self._execute_concurrently = concurrency_config.enabled and max_workers > 1
+
+        self._thread_pool_executor = (
+            ThreadPoolExecutor(
+                max_workers=min(
+                    concurrency_config.max_database_query_concurrency,
+                    max_workers,
+                )
+            )
+            if self._execute_concurrently
+            else None
+        )
+
+    def __exit__(self, exc_type, exc_value, traceback):
+        self.shutdown()
+        # Do not do use the context manager exception arguments, in order to get the desired default behavior where any
```

desired behavior == does not catch or block exemptions?

---

Desired behavior is to not catch, not block, nor suppress any exceptions. I updated the comment to more explicitly state this.